browser/resources/brave_new_tab_page_refresh/components/app.style.ts

```diff
@@ -162,29 +162,54 @@ export const style = scoped.css`
 
   .widget-container {
     --widget-height: 128px;
-    --widget-width: 450px;
-    --widget-flex-basis: var(--widget-width);
+    --widget-min-width: 380px;
+    --widget-max-width: 512px;
+    --widget-count: 0;
+    --widget-gap: 16px;
+    --widget-gap-total:
+      calc((var(--widget-count) - 1) * var(--widget-gap));
+    --widget-available-width: calc(
+      100cqi / var(--widget-count) -
+      var(--widget-gap-total) / var(--widget-count));
+    --widget-flex-basis: max(
```

Comment:
These CSS calculations are used to ensure that all widgets have the same width, while also being able to vary between a min and max width based on the container size.